<?xml version="1.0"?>
<launch>
  
  <!-- start effort controllers -->
  <include file="$(find grips_control)/launch/grips_effort_controllers.launch"/>
  
  <!-- PyKDL Jacobian Transpose Cartesian Controller -->
  <node name="jt_cartesian_controller" pkg="grips_control" type="jt_cartesian_controller.py" output="screen">
    <rosparam command="load" file="$(find grips_control)/config/jt_cartesian_controller.yaml"/>
  </node>
  
    <!-- State of the endpoint (FK) -->
  <node name="endpoint_state" pkg="grips_control" type="endpoint_state.py" >
    <rosparam command="load" file="$(find grips_control)/config/jt_cartesian_controller.yaml"/>
  </node>
  
  <!-- Initial position -->
  <node name="initial_xyz_pos" pkg="rostopic" type="rostopic" args="pub -1 /grips/ik_command geometry_msgs/PoseStamped '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: base_link}, pose: { position: { x: 0.25, y: 0.9, z: 0.5 }, orientation: {x: -0.7071, y: 0, z: 0, w: 0.7071 } } }'" />

</launch>
